osquery/tables/system/windows/windows_crashes.cpp

```diff
@@ -0,0 +1,672 @@
+                                         sizeof(context),
+                                         nullptr,
+                                         nullptr,
+                                         0,
+                                         nullptr) == S_OK) {
+    advanced->SetThreadContext(&context, sizeof(context));
+  }
+
+  auto status = registers->OutputRegisters(DEBUG_OUTCTL_THIS_CLIENT,
+                                           DEBUG_REGISTERS_DEFAULT);
+  client->SetOutputCallbacks(nullptr);
+  return (status == S_OK) ? Status() : Status(1);
+}
+
+Status logPEPathAndVersion(IDebugSymbols3* symbols, Row& r) {
+  char pePath[MAX_PATH + 1] = {0};
+  if (symbols->GetModuleNameString(
+          DEBUG_MODNAME_IMAGE, 0, 0, pePath, MAX_PATH + 1, nullptr) == S_OK) {
+    r["path"] = pePath;
+  }
+
+  VS_FIXEDFILEINFO version = {0};
+  if (symbols->GetModuleVersionInformation(
+          0, 0, "\\", &version, sizeof(version), nullptr) == S_OK) {
+    std::ostringstream versionStr;
+    versionStr << ((version.dwFileVersionMS >> 16) & 0xffff) << "."
+               << ((version.dwFileVersionMS >> 0) & 0xffff) << "."
+               << ((version.dwFileVersionLS >> 16) & 0xffff) << "."
+               << ((version.dwFileVersionLS >> 0) & 0xffff);
+    r["version"] = versionStr.str();
+    return Status();
+  }
+  return Status(1);
+}
+
+Status logModulePath(IDebugSymbols3* symbols, Row& r) {
+  std::istringstream converter(r["exception_address"]);
+  unsigned long long exAddr;
+  converter >> std::hex >> exAddr;
+  if (converter.fail()) {
+    return Status(1);
+  }
+
+  unsigned long modIndex = 0;
+  char modPath[MAX_PATH + 1] = {0};
+  if ((symbols->GetModuleByOffset(exAddr, 0, &modIndex, nullptr) == S_OK) &&
+      (symbols->GetModuleNameString(
+           DEBUG_MODNAME_IMAGE, modIndex, 0, modPath, MAX_PATH + 1, nullptr) ==
+       S_OK)) {
+    r["module"] = modPath;
+    return Status();
+  }
+  return Status(1);
+}
+
+Status logPEBInfo(IDebugClient5* client,
+                  IDebugControl5* control,
+                  IDebugSymbols3* symbols,
+                  IDebugSystemObjects2* system,
+                  IDebugDataSpaces4* data,
+                  Row& r) {
+  // Get ntdll symbols
+  symbols->Reload("/f ntdll.dll");
+  unsigned long long ntdllBase = 0;
+  if (symbols->GetModuleByModuleName("ntdll", 0, nullptr, &ntdllBase) != S_OK) {
+    return Status(1);
+  }
+
+  // Get PEB address
+  unsigned long long pebAddr = 0;
+  if (system->GetCurrentProcessPeb(&pebAddr) != S_OK) {
+    return Status(1);
+  }
+
+  // Get ProcessParameters offset in the PEB
+  unsigned long pebTypeId = 0;
+  unsigned long procParamsOffset = 0;
+  if ((symbols->GetTypeId(ntdllBase, "_PEB", &pebTypeId) != S_OK) ||
+      (symbols->GetFieldOffset(
+           ntdllBase, pebTypeId, "ProcessParameters", &procParamsOffset) !=
+       S_OK)) {
+    return Status(1);
+  }
+  // Get address of ProcessParameters struct
```

Comment:
nit: there's some inconsistency with newline spacing between your chunks of code here.